core/src/test/java/com/graphhopper/routing/util/CustomModelTest.java

```diff
@@ -18,94 +18,66 @@
 
 package com.graphhopper.routing.util;
 
+import com.graphhopper.json.Clause;
 import org.junit.jupiter.api.Test;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.Iterator;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class CustomModelTest {
 
-    static CustomModel setValue(CustomModel model, String op, String encodedValue, double value) {
-        Map<String, Object> map = new HashMap<>();
-        map.put(op + value, 0);
-        model.getPriority().put(encodedValue, map);
-        return model;
-    }
-
-    Object getValue(CustomModel model, String encodedValue) {
-        Map map = (Map) model.getPriority().get(encodedValue);
-        if (map == null || map.isEmpty())
-            return null;
-        return map.keySet().iterator().next();
+    @Test
+    public void testTooBigFactor() {
+        CustomModel truck = new CustomModel();
+        truck.getPriority().add(Clause.createIf("max_width < 3", 10));
```

Comment:
Can we use shorter method names here: `Clause.if` instead of `Clause.createIf` etc.? Since this will be used a lot in a rather repetitive way when setting up custom models via Java shorter is better here IMO.

Reply:
This is only this ugly because `if` is not possible as a method name (`if` is a keyword). Also e.g. `new IfClause` is not much shorter but could be better because you can use autocomplete for the rest?

Reply:
Oh.. right ;) Maybe `Clause.ifClause`? Then with static import for `Clause` we can write: `getPriority().add(ifClause("max_width < 3", 10));` But if we merge the speed sections we also need to pass the operator here?

Reply:
`Clause.cond` ?

Reply:
You mean 

```java
Clause.cond("if", "max_width < 3", "multiply_with", 0.8);
Clause.cond("else if", ""max_width < 5", multiply_with", 0.7);
Clause.cond("else", "", multiply_with", 0.7);
```

I think in this case I would just use `new Clause(...)`?

Reply:
Do we even need to distinguish three cases? `else if` is just syntactic sugar

```java
if (cond1) {
} else if (cond2) {
} else if (cond3) {
} else {
}
```
is equivalent to:
```java
if (cond1) {
} else {
   if (cond2) {
   } else {
      if (cond3) {
      } else {
      }
   }
}
```
So we would end up with:

```java
Clause.cond("max_width < 3", "multiply_with", 0.8)
Clause.cond("max_width < 5", "multiply_with", 0.7);
Clause.cond("multiply_with", 0.7);
```

Reply:
> Maybe Clause.ifClause? 

Or `Clause.IF` ;) or `Clause.ifThen`?

> Clause.cond ?
> I think in this case we could just as well use new Clause(...)?

Currently I tend towards `new IfClause()` or static import and some short method name. 

Another idea is to create static methods ala:
```
Clause.addIf(model.getPriority(), "max_width < 3", "multiply_with", 0.8);
```

IMO it would be nice if it also would be chainable (and make the operator less dynamic?):

```
Clause.addIf(model.getPriority(), "max_width < 3", MultiplyOp.class, 0.8).
       addElse(MultiplyOp.class, 0.4)
```

or something.

Reply:
> IMO it would be nice if it also would be chainable (and make the operator less dynamic?):
> 
> ```
> Clause.addIf(model.getPriority(), "max_width < 3", MultiplyOp.class, 0.8).
>        addElse(MultiplyOp.class, 0.4)
> ```
> 
> or something.

A StepBuilder would be nice.

Reply:
> is equivalent to:

We do not support nested if blocks

> Or Clause.IF ;) or Clause.ifThen?

Yes why not uppercase? Maybe `Clause.If` to distinguish the method from some static constant (which would normally be all uppercase)?. I don't like `ifThen` as much because it looks like there is something more going on when really all we do is create a data class.

> new IfClause()

Ok

> Another idea is to create static methods ala:

Why would this be better?

> IMO it would be nice if it also would be chainable

This can be achieved with a simple helper method in custom model:

```java
CustomModel addPriority(Clause clause) {
    getPriority().add(clause);
    return this;
}
```

> and make the operator less dynamic

What do you mean? And why .class in `MultiplyOp.class`? I thought we could just use static constants or an enum for the operator name?

Reply:
> Maybe Clause.If to distinguish the method from some static constant

Good idea :+1: 

> This can be achieved with a simple helper method in custom model:

:+1: 

> What do you mean? And why .class in MultiplyOp.class? I thought we could just use static constants or an enum for the operator name?

Yes, static constants or enum is better. Just so that we do not have to correctly spell strings and maybe introduce operator specific functionality or similar.